euouae_check.py

```diff
@@ -0,0 +1,88 @@
+# Accepts as arguments file or directory paths,
+# checks all gabc files containing antiphons
+# for standard EUOUAE lyrics and tones.
+
+import os
+import os.path
+import sys
+
+import arpeggio
+from music21 import converter, volpiano
+from music21.text import assembleLyrics
+import chant21
+from chant21.gabc.converter import AlterationWarning
+
+euouae_tones = {}
+
+
+class Skip(RuntimeError):
+    pass
+
+
+def _check(filename, verbose):
+    if not filename.endswith('.gabc'):
+        raise Skip('not a gabc')
+    if '_' in filename:
+        raise Skip('submitter (not selected) version')
+
+    if verbose:
+        print('checking {}'.format(filename))
+
+    try:
+        chant = converter.parse(filename)
+    except (arpeggio.NoMatch, AlterationWarning):
+        raise Skip('chant21 cannot parse')
+
+    metadata = chant.editorial.metadata
+    office_part = metadata.get('office-part')
+    if office_part != 'an':
+        raise Skip('not an antiphon ({})'.format(office_part))
+
+    errors = []
+
+    last_section = chant.sections[-1]
+    lyrics = assembleLyrics(last_section)
+    has_euouae = 'E U O U A E.' in lyrics
```

Comment:
Ideally this would check If the EUOUAE is all caps, and if the final E has a dot after, instead of only treating its melody if it matches the "E U O U A E." format.